src/environments/basics/TwoDimensionalNonlinearPolynomialEnv.jl

```diff
@@ -10,9 +10,11 @@ end
 
 function dynamics!(env::TwoDimensionalNonlinearPolynomialEnv)
     return function (dx, x, p, t; u)
+        @assert length(u) == 1
+        _u = u[1]  # Real or Array
```

Comment:
Why did you do this?

Reply:
TL; DR) For convenient usage (while losing type stability).

In Julia, the types of `1.0` (`Number`) and `[1]` (`Array`) are different.
The dynamical equations of the system considered are based on inputs whose type is `Number`.
On the other hand, the optimisation solver (`NLopt`) produces optimizer as an array whose length is `1`.